apis/PostsEndpoint.js

```diff
@@ -56,30 +60,19 @@ const formatPostData = (data)=> {
   return postData;
 };
 
-const parseBody = (res, apiResponse, req, method) => {
-  const { body } = res;
-
-  if (method === 'get') {
-    const { data } = body;
-
-    if (data && data.children && data.children[0]) {
-      if (data.children.length === 1) {
-        apiResponse.addResult(PostModel.fromJSON(data.children[0].data));
-        return;
-      } else {
-        data.children.forEach(c => apiResponse.addResult(PostModel.fromJSON(c.data)));
-        return;
-      }
-    } else if (data) {
-      return;
-    }
-  } else if (method !== 'del') {
-    if (body.json && body.json.errors.length === 0) {
-      apiResponse.addResult(body.json.data);
+const handleGet = (res, apiResponse, req, method) => {
+  const { body: { data } } = res;
+
+  if (data && data.children && data.children[0]) {
+    if (data.children.length === 1) {
+      apiResponse.addResult(PostModel.fromJSON(data.children[0].data));
       return;
     } else {
-      throw body.json;
+      data.children.forEach(c => apiResponse.addResult(PostModel.fromJSON(c.data)));
+      return;
     }
+  } else if (data) {
```

Comment:
that was in the GET block in the previous code so I didn't want to mess with it. I have no idea what it's intended for though.